src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -83,17 +86,15 @@ public void execute(final MavenProject project) {
                     Files.write(DetectorPathManager.originalOrderPath(), DetectorPlugin.getOriginalOrder(project));
                 }
 
-                minimizedResults()
-                        // Nothing to fix if the order is already passing
-                        // TODO: Make sure we don't break the passing order though
-                        .filter(minimized -> !minimized.expected().equals(Result.PASS))
-                        .forEach(minimized -> {
-                            try {
-                                setupAndApplyFix(minimized);
-                            } catch (Exception e) {
-                                throw new RuntimeException(e);
-                            }
-                        });
+                Map<String, List<MinimizeTestsResult>> dependentTestResults = minimizedResults()
+                        .collect(Collectors.groupingBy(MinimizeTestsResult::dependentTest));
+                for (Map.Entry<String, List<MinimizeTestsResult>> entry : dependentTestResults.entrySet()) {
+                    try {
+                        setupAndApplyFix(entry.getValue());
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
```

Comment:
I don't think we need to rethrow this as a `RuntimeException` because it's inside a `Callable` (and this way there's less visual overhead with wrapping the `Exception`).